lms/services/exceptions.py

```diff
@@ -218,3 +218,15 @@ class CanvasFileNotFoundInCourse(ServiceError):
     def __init__(self, file_id):
         self.details = {"file_id": file_id}
         super().__init__(self.details)
+
+
+class HTTPError(ServiceError):
+    """A problem with an HTTP request to an external service."""
+
+    def __init__(self, response=None):
+        super().__init__(response)
+        self.response = response
+
+
+class HTTPValidationError(HTTPError):
+    """An invalid HTTP response from an external service."""
```

Comment:
We normally define our `<APP>/<FOO>/exceptions.py` files, rather than in the individual `services/http.py` file. This is an attempt to achieve a balance between exception specificity versus exception reusability.
See https://stackoverflow.com/c/hypothesis/questions/295. In this case these two exceptions are currently only raised by `HTTPService`, but I put them here anyway because 1. It's where put put all our services exceptions; 2. It encourages re-use, if we have another HTTP-related service in future.